CMakeLists.txt

```diff
@@ -205,6 +207,98 @@ if (NN_ENABLE_NANOCAT)
     target_link_libraries (nanocat ${PROJECT_NAME})
 endif ()
 
+find_program (ASCIIDOCTOR_EXE asciidoctor)
```

Comment:
Psuedocode, starting at line 221 below:

IF NN_ENABLE_DOC
check for asciidoctor
if asciidoctor doesn't exist, fatal error
ELSEIF (not NN_ENABLE_DOC)
# do nothing; no check for asciidoctor needed, since docs not requested